tee-worker/litentry/core/evm-dynamic-assertions/src/lib.rs

```diff
@@ -0,0 +1,296 @@
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+extern crate alloc;
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+// re-export module to properly feature gate sgx and regular std environment
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+pub mod sgx_reexport_prelude {
+	pub use hex_sgx as hex;
+	pub use http_sgx as http;
+}
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+use crate::sgx_reexport_prelude::*;
+
+use crate::precompiles::Precompiles;
+use ethabi::{
+	decode, encode,
+	ethereum_types::{H160, U256},
+	ParamType, Token,
+};
+use evm::{
+	backend::{MemoryBackend, MemoryVicinity},
+	executor::stack::{MemoryStackState, StackExecutor, StackSubstateMetadata},
+	Config,
+};
+use lc_dynamic_assertion::{
+	AssertionExecutor, AssertionLogicRepository, AssertionResult, Identity, IdentityNetworkTuple,
+	Web3Network,
+};
+use std::{
+	collections::BTreeMap,
+	string::{String, ToString},
+	sync::Arc,
+	vec,
+	vec::Vec,
+};
+
+mod precompiles;
+pub mod repository;
+pub mod sealing;
+
+// feature guard ?
+pub mod mock;
+
+pub use itp_settings::files::ASSERTIONS_FILE;
+
+pub type AssertionId = H160;
+pub type SmartContractByteCode = Vec<u8>;
+pub type AssertionRepositoryItem = (SmartContractByteCode, Vec<String>);
+
+pub struct EvmAssertionExecutor<A: AssertionLogicRepository> {
+	pub assertion_repository: Arc<A>,
+}
+
+impl<A: AssertionLogicRepository<Id = H160, Item = AssertionRepositoryItem>>
+	AssertionExecutor<AssertionId> for EvmAssertionExecutor<A>
+{
+	fn execute(
+		&self,
+		assertion_id: A::Id,
+		identities: &[IdentityNetworkTuple],
+	) -> Result<AssertionResult, String> {
+		let (smart_contract_byte_code, secrets) = self
+			.assertion_repository
+			.get(&assertion_id)
+			.map_err(|_| "Could not access assertion repository")?
+			.ok_or("Assertion not found")?;
+		let input = prepare_execute_call_input(identities, secrets)
+			.map_err(|_| "Could not prepare evm execution input")?;
+
+		// prepare EVM runtime
+		let config = prepare_config();
+		let vicinity = prepare_memory();
+		let state = BTreeMap::new();
+		let mut backend = MemoryBackend::new(&vicinity, state);
+		let metadata = StackSubstateMetadata::new(u64::MAX, &config);
+		let state = MemoryStackState::new(metadata, &mut backend);
+		let precompiles = Precompiles {};
+		let mut executor = StackExecutor::new_with_precompiles(state, &config, &precompiles);
+
+		// caller, just an unused account
+		let caller = hash(5); //0x05
+
+		// deploy assertion smart contract
+		let address = executor.create_address(evm::CreateScheme::Legacy { caller });
+		let _create_result = executor.transact_create(
+			caller,
+			U256::zero(),
+			smart_contract_byte_code,
+			u64::MAX,
+			Vec::new(),
+		);
+
+		// call assertion smart contract
+		let call_result =
+			executor.transact_call(caller, address, U256::zero(), input, u64::MAX, Vec::new());
```

Comment:
I'm not sure if this is a real concern, but will the `gas limit` be enough to prevent the smart contracts (untrusted code) from doing something malicious

Reply:
This only prevents from infinite execution.